services/blockchain-indexer/shared/dataService/business/transactionsEstimateFees.js

```diff
@@ -244,6 +246,36 @@ const calcAdditionalFees = async (transaction) => {
 	return additionalFees;
 };
 
+const getTransactionParamsSchema = (transaction, metadata) => {
+	const moduleMetadata = metadata.modules.find(m => m.name === transaction.module);
+	const { params: schema } = moduleMetadata.commands.find(c => c.name === transaction.command);
+	return schema;
+};
+
+const validateTransactionParams = async transaction => {
+	// Mock optional values if not present before schema validation.
+	if (transaction.module === MODULE.TOKEN
+		&& transaction.command === COMMAND.TRANSFER_CROSS_CHAIN) {
+		if (!('messageFee' in transaction.params)) {
+			transaction.params.messageFee = '10000000';
+		}
+
+		if (!('messageFeeTokenID' in transaction.params)) {
+			transaction.params.messageFeeTokenID = '0000000000000000';
+		}
+	}
+
+	const metadata = await getSystemMetadata();
+	const txParamsSchema = getTransactionParamsSchema(transaction, metadata);
+	const parsedTxParams = parseInputBySchema(transaction.params, txParamsSchema);
```

Comment:
I am not sure but i think we don't need to parse params. we can directly pass it to validator.